docs/design/design_v1.9.md

```diff
@@ -69,196 +67,246 @@ This means we aim to standardize:
+- If authorization mode is WebHook, [Error] if webhook_authz.conf does not exists.
+- [Error] if advertise address is ipv6 and /proc/sys/net/bridge/bridge-nf-call-ip6tables does not exists/does not contains 1.
 
 Please note that:
 
-1. Preflight checks can be invoked individually with the `kubeadm phase preflight` command.
+1. Preflight checks can be invoked individually with the `kubeadm alpha phase preflight` command.
 
 ### Generate the necessary certificates
 
 `kubeadm` generates certificate and private key pairs for different purposes.
 Certificates are stored by default in `/etc/kubernetes/pki`. This directory is configurable.
 
 There should be:
- - a CA certificate (`ca.crt`) with its private key (`ca.key`)
- - an API Server certificate (`apiserver.crt`) using `ca.crt` as the CA with its private key (`apiserver.key`). The certificate should:
-   - be a serving server certificate (`x509.ExtKeyUsageServerAuth`)
-   - contain altnames for
-     - the kubernetes' service's internal clusterIP and dns name (e.g. `10.96.0.1`, `kubernetes.default.svc.cluster.local`, `kubernetes.default.svc`, `kubernetes.default`, `kubernetes`)
-     - the node-name
-     - the IPv4 address of the default route
-     - optional extra altnames that can be specified by the user
- - a client certificate for the apiservers to connect to the kubelets securely (`apiserver-kubelet-client.crt`) using `ca.crt` as the CA with its private key (`apiserver-kubelet-client.key`). The certificate should:
-   - be a client certificate (`x509.ExtKeyUsageClientAuth`)
-   - be in the `system:masters` organization
- - a private key for signing ServiceAccount Tokens (`sa.key`) along with its public key (`sa.pub`)
- - a CA for the front proxy (`front-proxy-ca.crt`) with its key (`front-proxy-ca.key`)
- - a client cert for the front proxy client (`front-proxy-client.crt`) using `front-proxy-ca.crt` as the CA with its key (`front-proxy-client.key`)
+ - A CA certificate (`ca.crt`) with its private key (`ca.key`).
+ - An API server certificate (`apiserver.crt`) using `ca.crt` as the CA with its private key (`apiserver.key`). The certificate should:
+   - Be a serving server certificate (`x509.ExtKeyUsageServerAuth`).
+   - Contains altnames for:
+     - The Kubernetes service's internal clusterIP (the first address in the services CIDR, e.g. `10.96.0.1` if service subnet is `10.96.0.0/12`).
+     - Kubernetes DNS names (e.g.  `kubernetes.default.svc.cluster.local` if `--service-dns-domain` is `cluster.local`, `kubernetes.default.svc`, `kubernetes.default`, `kubernetes`).
+     - The node-name.
+     - The `--apiserver-advertise-address`.
+     - Optional extra altnamesspecified by the user.
+ - A client certificate for the apiservers to connect to the kubelets securely (`apiserver-kubelet-client.crt`) using `ca.crt` as the CA with its private key (`apiserver-kubelet-client.key`). The certificate should:
+   - Be with `kube-apiserver-kubelet-client` CN.
```

Comment:
this is not a requirement; can you separate between requirements (clientcert and system:masters) and what kubeadm additionally does (the CN)